tools/kotlin-native-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KonanCompileTask.kt

```diff
@@ -201,6 +201,17 @@ open class KonanCompileConfig(
 
     // DSL. Libraries.
 
+    fun library(project: Project) = {
+        this.project.evaluationDependsOn(project.path)
+        compilationTask.dependsOn(project.tasks.findByPath("compileKonan"))
+        libraries(project.fileTree(project.konanCompilerOutputDir).include("*.klib"))
+    }
+    fun library(project: Project, artifactName: String) = {
+        this.project.evaluationDependsOn(project.path)
+        compilationTask.dependsOn(project.konanArtifactsContainer.getByName(artifactName).compilationTask)
+        library("${project.konanCompilerOutputDir}/$artifactName.klib")
```

Comment:
The output file name of the library may be changed by user via DSL and this path may become incorrect. The compilation task you got via the `artifactName` has a property `artifact` which can be used here.

Reply:
Also we need to check if the artifact we're trying to add as a library is really klib or not (see the  `produce` property of the compilation task).